app/server/BeelieversAuction/auth.server.ts

```diff
@@ -0,0 +1,140 @@
+import { SuiClient, type SuiTransactionBlockResponse } from "@mysten/sui/client";
+
+export type BidDetails = {
+	sender: string;
+	auctionId: string;
+	totalBidAmount: string;
+};
+
+type TxData = {
+	effects?: {
+		status?: {
+			status?: "success" | "failure";
+			error?: string;
+		};
+	} | null;
+	events?:
+		| {
+				type: string;
+				parsedJson: unknown;
+				sender: string;
+		  }[]
+		| null;
+};
+
+type JsonRpcResponse = {
+	jsonrpc: string;
+	id: number;
+	result?: TxData;
+};
+
+//TODO: move it to env, or controller constructor
+const TRUSTED_PACKAGE_ID = "0xd5b24b83b168f8656aa7c05af1256e6115de1b80d97be0cddf19297a15535149";
+
+function processTransactionData(
+	data: TxData | undefined,
+	suiTxId: string,
+	bidderAddr: string,
+	source: "Primary" | "Fallback",
+): BidDetails | null {
```

Comment:
instead of doing console.error here all the time, let's return `"<error message>"`. And we can log in `controller`. So the return type of this function should be `BidDetails |TxCheckError`

Reply:
```suggestion
): BidDetails | TxCheckError {
```